pkg/config/config.go

```diff
@@ -42,6 +47,45 @@ func LoadIPAMConfig(bytes []byte, envArgs string) (*types.IPAMConfig, string, er
 		return nil, "", fmt.Errorf("LoadArgs - CNI Args Parsing Error: %s", err)
 	}
 
+	// Once we have our basics, let's look for our (optional) configuration file
+	confdirs := []string{"/etc/kubernetes/cni/net.d/whereabouts.d/whereabouts.conf", "/etc/cni/net.d/whereabouts.d/whereabouts.conf"}
+	// We prefix the optional configuration path (so we look there first)
+	if n.IPAM.ConfigurationPath != "" {
+		confdirs = append([]string{n.IPAM.ConfigurationPath}, confdirs...)
+	}
+
+	// Cycle through the path and parse the JSON config
+	flatipam := types.Net{}
+	foundflatfile := ""
+	for _, confpath := range confdirs {
+		if pathExists(confpath) {
+
+			jsonFile, err := os.Open(confpath)
+
+			if err != nil {
+				return nil, "", fmt.Errorf("Error opening flat configuration file @ %s with: %s", confpath, err)
+			}
+
+			defer jsonFile.Close()
+
+			jsonBytes, _ := ioutil.ReadAll(jsonFile)
```

Comment:
Uncaught error; I'd return it.

Reply:
Thanks for the eye, good call, got it in the most recent commit